packages/config/src/reader.ts

```diff
@@ -83,9 +83,27 @@ export class ConfigReader implements Config {
     // Merge together all configs into a single config with recursive fallback
     // readers, giving the first config object in the array the lowest priority.
     return configs.reduce<ConfigReader>(
-      (previousReader, { data, context, filteredKeys }) => {
+      (previousReader, { data, context, filteredKeys, deprecatedKeys }) => {
         const reader = new ConfigReader(data, context, previousReader);
         reader.filteredKeys = filteredKeys;
+
+        // TODO(cmpadden) introduce `deprecatedKeys` and `notifiedDeprecatedKeys` & use logger
```

Comment:
@freben , the user is warned that they are accessing a key present in `filteredKeys` only once via `getOptional`. Should similar logic be added for the `deprecatedKeys`? Or should we inform the user of all deprecated keys in use on load?

As an aside, when I introduced a `reader.deprecatedKeys` in the `reduce` method, it was ultimately undefined because of how it is being assigned to the reader while `previousReader` is undefined. This is why I temporarily threw the `console.warn`s in here. Still trying to figure that one out!

Reply:
I think this is alright tbh, seems like a good place to warn that works in for both frontend and backend, development and production. I would add something to `config:check` perhaps though, probably even fail the check if deprecated config is used

Reply:
That's a great idea, thank you.

I've introduced `backstage-cli config:check --deprecated`